src/packages/client/src/__tests__/integration/happy/native-types-mysql/test.ts

```diff
@@ -0,0 +1,210 @@
+import { getTestClient } from '../../../../utils/getTestClient'
+import path from 'path'
+import { migrateDb } from '../../__helpers__/migrateDb'
+import Decimal from 'decimal.js'
+
+beforeAll(async () => {
+  process.env.TEST_MYSQL_URI += '-native-types-tests'
+  await migrateDb({
+    connectionString: process.env.TEST_MYSQL_URI!,
+    schemaPath: path.join(__dirname, 'schema.prisma'),
+  })
+})
+
+test('native-types-mysql A: Int, SmallInt, TinyInt, MediumInt, BigInt', async () => {
+  const PrismaClient = await getTestClient()
+
+  const prisma = new PrismaClient()
+
+  await prisma.a.deleteMany()
+
+  const data = {
+    int: 123,
+    sInt: 12,
+    tInt: true,
+    mInt: 100,
+    bInt: 123123123,
+  }
+
+  const e = await prisma.a.create({
+    data,
+    select: {
+      int: true,
+      sInt: true,
+      mInt: true,
+      bInt: true,
+      tInt: true,
+    },
+  })
+
+  expect(e).toEqual(data)
+
+  prisma.$disconnect()
+})
+
+test('native-types-mysql B: Float, Double, Decimal, Numeric', async () => {
+  const PrismaClient = await getTestClient()
+
+  const prisma = new PrismaClient()
+
+  await prisma.b.deleteMany()
+
+  let data: any = {
+    float: 12.2,
+    dfloat: 10.2,
+    decFloat: 1.1,
+    numFloat: '5.6',
+  }
+
+  let b = await prisma.b.create({
+    data,
+    select: {
+      float: true,
+      dfloat: true,
+      decFloat: true,
+      numFloat: true,
+    },
+  })
+
+  expect(Decimal.isDecimal(b.float)).toBe(false)
+  expect(Decimal.isDecimal(b.dfloat)).toBe(false)
+  expect(Decimal.isDecimal(b.decFloat)).toBe(true)
+  expect(Decimal.isDecimal(b.numFloat)).toBe(true)
+
+  expect(b).toMatchInlineSnapshot(`
+    Object {
+      decFloat: 1.1,
+      dfloat: 10.2,
+      float: 12.2,
+      numFloat: 5.6,
+    }
+  `)
+
+  data = {
+    float: 12.2,
+    dfloat: 10.2,
+    decFloat: new Decimal(1.1),
+    numFloat: new Decimal('5.6'),
+  }
+
+  b = await prisma.b.create({
+    data,
+    select: {
+      float: true,
+      dfloat: true,
+      decFloat: true,
+      numFloat: true,
+    },
+  })
+
+  expect(data).toEqual(b)
+
+  prisma.$disconnect()
+})
+
+test('native-types-mysql C: Char, VarChar, TinyText, Text, MediumText, LongText', async () => {
+  const PrismaClient = await getTestClient()
+
+  const prisma = new PrismaClient()
+
+  await prisma.c.deleteMany()
+
+  const data = {
+    char: 'f0f0f0f0f0',
+    vChar: '12345678901',
+    tText: 'foo',
+    text: 'txt ➡️',
+    mText: '🥳',
+    ltext: '🔥',
```

Comment:
Maybe we could have longer texts here?

For example
- `text` could have 300 chars (more than tinytext)
- `mText` could have 70k chars (more than text)
- `lText` could have 17000k chars (more than mediumtext)

Using a programmatic way like `'Something'.repeat(1000)`

If it's too much maybe we can do it only for text?

```
      Type | Maximum length
-----------+-------------------------------------
  TINYTEXT |           255 (2 8−1) bytes
      TEXT |        65,535 (216−1) bytes = 64 KiB
MEDIUMTEXT |    16,777,215 (224−1) bytes = 16 MiB
  LONGTEXT | 4,294,967,295 (232−1) bytes =  4 GiB
```